backend/core/src/main/java/org/sonarsource/sonarlint/core/tracking/IssueMatchingService.java

```diff
@@ -157,6 +290,68 @@ private void refreshServerIssues(SonarLintCancelMonitor cancelMonitor, Binding b
     CompletableFuture.allOf(fetchTasks.toArray(new CompletableFuture[0])).join();
   }
 
+  private List<TrackedIssue> newMatchIssues(Path serverRelativePath, List<ServerIssue<?>> serverIssues,
+    List<LocalOnlyIssue> localOnlyIssues, List<TrackedIssue> trackedIssues, NewCodeDefinition newCodeDefinition) {
+    var serverIssueMatcher = new IssueMatcher<>(new TrackedIssueFindingMatchingAttributeMapper(), new ServerIssueMatchingAttributesMapper());
+    var serverMatchingResult = serverIssueMatcher.match(trackedIssues, serverIssues);
+    var localIssueMatcher = new IssueMatcher<>(new TrackedIssueFindingMatchingAttributeMapper(), new LocalOnlyIssueMatchingAttributesMapper());
+    var localMatchingResult = localIssueMatcher.match(trackedIssues, localOnlyIssues);
```

Comment:
We are checking local matches here for the third time

Reply:
Correct, and it's also related to your first comment, we keep this for backward compatibility. It will change in the future